app/auth/Signin.tsx

```diff
@@ -1,34 +1,52 @@
 import React, { useState } from "react";
-import { View, StyleSheet, KeyboardAvoidingView, ScrollView, Platform } from "react-native";
+import { View, StyleSheet, KeyboardAvoidingView, ScrollView, Platform, Keyboard } from "react-native";
 import { router } from "expo-router";
+import { SigninI } from "@/interfaces";
 import { useAuthStore } from "@/store";
+import { useFormikHook } from "@/hooks";
+import { signinValidationSchema } from "@/utils";
 import { colorPalette, LayoutStyles, Spacing } from "@/styles";
 import { AppButton, AppLogo, AppText, Checkbox, GradientWrapper, TextInput } from "@/components";
 
 const Signin = () => {
   const { login } = useAuthStore();
   const [isFirstSignIn, setIsFirstSignIn] = useState<boolean>(true);
```

Comment:
This is wrong logic for checking isFirstSignin. This should be inside the auth store.